auth/token_authentication.py

```
@@ -98,3 +113,65 @@ def __call__(self, request: Request) -> ApiKey:
+
+    return token
+
+
+def verify_ephemeral_api_key(token: str) -> ApiKey:
+    signed_token = token.removeprefix(EPHEMERAL_KEY_PREFIX)
+    try:
+        payload = _signer.loads(
+            signed_token, salt=EMERPHAL_KEY_SALT, max_age=TOKEN_EXPIRATION
+        )
+    except SignatureExpired:
+        raise AuthenticationError("API Key has expired.")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)